src/pages/directory.tsx

```diff
@@ -0,0 +1,125 @@
+import { useState } from "react";
+import { DataGrid, type GridColDef } from "@mui/x-data-grid";
+import { api } from "../utils/api";
+import DefaultLoadingPage from "~/components/loading/loading";
+import ErrorPage from "../components/error/error";
+import UserForm, { type CreateUserData } from "~/components/UserForm";
+import { type SubmitHandler } from "react-hook-form";
+
+enum UserRole {
+  EMPTY = "",
+  MENTEE = "Mentee",
+  MENTOR = "Mentor",
+  ADMIN = "Admin",
+}
+
+const Directory = () => {
+  const utils = api.useUtils();
+  const { data, error, isLoading } = api.user.getAll.useQuery();
+  const [selectionModel, setSelectionModel] = useState<string[]>([]);
+  const createUserMutation = api.user.createUser.useMutation({
+    onSuccess: () => {
+      void utils.user.getAll.invalidate();
+    },
+  });
+  const deleteUsersMutation = api.user.deleteByIds.useMutation({
+    onSuccess: () => {
+      void utils.user.getAll.invalidate();
+    },
+  });
+
+  const onSubmit: SubmitHandler<CreateUserData> = async (data) => {
+    await createUserMutation.mutateAsync({
+      ...data,
+      role: data.role.toUpperCase(),
+    });
+  };
+
+  const handleDeleteUsers = () => {
+    if (selectionModel.length > 0) {
+      deleteUsersMutation
+        .mutateAsync({ ids: selectionModel })
+        .then(() => {
+          setSelectionModel([]);
+        })
+        .catch((error) => {
+          console.error("Failed to delete users:", error);
```

Comment:
Is there a confirmation to delete users? If not we should probably have one - could be pretty damaging if done by accident. Also might mitigate prevent an error if the delete request goes through twice and the second one errors